core/network/src/protocol/sync.rs

```diff
@@ -847,10 +896,14 @@ impl<B: BlockT> ChainSync<B> {
 		true
 	}
 
+	/// Convenience function to iterate through all peers and see if there are any that we are
+	/// downloading this hash from.
 	fn is_already_downloading(&self, hash: &B::Hash) -> bool {
 		self.peers.iter().any(|(_, p)| p.state == PeerSyncState::DownloadingStale(*hash))
 	}
 
+	/// Convenience function to check if the given hash exists in our block queue or the provided
+	/// chain in `protocol` where the status isn't Unknown.
```

Comment:
/// Returns true if the block exists in the import queue or is already imported.